### ISSUE TITLE
Remove `role=button` from button links

### ISSUE THREAD

**Issue by manuelpuyol** [primer/view_components#879]
[`Primer::ButtonComponent`](https://primer.style/view-components/components/button) adds `role="button"` when the component is rendering as an `<a>` tag. This causes screenreaders to announce what should be a link as a button.

https://github.com/primer/view_components/blob/847c1718277387cb9889a4f71505034ea671afa1/app/components/primer/base_button.rb#L31-L35

**Comment by owenniblock** [comment]
I see we have 3 options for the `:tag` (`:button`, `:a` or `:summary`).

I'll also remove the `role="button"` from the summary tag: https://www.w3.org/TR/html-aria/#be-cautious-of-side-effects
